src/labone/mock/hpk_mock_server.py

```diff
@@ -0,0 +1,137 @@
+from abc import ABC
+from contextlib import contextmanager
+import json
+
+from labone.core import ListNodesFlags, ListNodesInfoFlags
+from labone.core.helper import LabOneNodePath
+from labone.core.session import NodeInfo
+from labone.core.subscription import StreamingHandle
+from labone.core.value import AnnotatedValue
+from labone.mock.mock_server import ConcreteMockServer
+from labone.mock.hpk_functionality import HpkMockFunctionality
+
+
+HPK_SCHEMA_ID = 11970870220622790664
+SERVER_ERROR = "SERVER_ERROR"
+
+
+
+def build_capnp_error(error: Exception):
+    """
+    Returns:
+        Capnp Type dictionary for: Result(Error)
+    """
+    return {
+        "err": {
+            "code": 2,
+            "message": f"{error}",
+            "category": SERVER_ERROR,
+            "source": __name__,
+        }
+    }
+
+
+class HpkMockServer(ConcreteMockServer):
+    id = HPK_SCHEMA_ID
+    name = "HPK"
+
+    def __init__(self, functionality: HpkMockFunctionality) -> None:
+        super().__init__()
+        self._functionality = functionality
+
+    async def getTheSchema(self, _context, **kwargs):
+        """Redirect to server method."""
+        return await self._base_server.getTheSchema(_context, **kwargs)
+
+    async def listNodes(self, pathExpression, flags, client, _context, **kwargs):
+        return await self._functionality.list_nodes(pathExpression, flags=flags)
+
+    async def listNodesJson(self, pathExpression, flags, client, _context, **kwargs):
+        return json.dumps(
+            await self._functionality.list_nodes_info(
+                path=pathExpression,
+                flags=flags,
+            )
+        )
+
+    async def getValue(
+        self,
+        pathExpression,
+        lookupMode,
+        flags,
+        client,
+        _context,
+        **kwargs,
+    ):
+        try:
+            if lookupMode == 0:  # direct lookup
+                responses = [await self._functionality.get(pathExpression)]
+            else:
+                responses = await self._functionality.get_with_expression(
+                    pathExpression, flags=flags
+                )
+        except BaseException as e:
+            return [build_capnp_error(e)]
+
+        return [
+            {
+                "ok": {
+                    "value": {"int64": response.value},
```

Comment:
issue: You explicitly allow only integer nodes with this! This already exists in `labone.core.value._value_from_python_types`

Reply:
`_value_from_python_types` needs the reflection server as a second argument. Can you tell me how to extract it from our existing fields?

Reply:
I wrote my own version of the function, not depending on anything like that. Please comment on it. If we want to use the already existing function, we have to passing something from the `MockServer` into the `SessionMockTemplate`, which is not desired

Reply:
We should stick to the current architecture. All conversions between the value in python and capnp should happen inside `labone.core.value`. Otherwise, it will be a nightmare to maintain. 

either you move your implementation int `labone.core.value` and name it `_value_from_python_types_dict` 
or you pass to the constructor of SessionMockTemplate the parsed schema. The second would be the cleaner solution. (Parse the schema outside of `SessionMockTemplate` and `MockServer` and pass it to both during construction.)